src/integrations/prefect-aws/prefect_aws/observers/ecs.py

```diff
@@ -414,15 +496,99 @@ async def replicate_ecs_event(event: dict[str, Any], tags: dict[str, str]):
+
+    task_arn = event.get("detail", {}).get("taskArn")
+    if not task_arn:
+        handler_logger.debug("No task ARN in event. Skipping.")
+        return
+
+    flow_run_id = tags.get("prefect.io/flow-run-id")
+
+    async with prefect.get_client() as orchestration_client:
+        try:
+            flow_run = await orchestration_client.read_flow_run(
+                flow_run_id=uuid.UUID(flow_run_id)
+            )
+        except ObjectNotFound:
+            logger.debug(f"Flow run {flow_run_id} not found, skipping")
+            return
+
+        assert flow_run.state is not None, "Expected flow run state to be set"
+
+        # Exit early for final or scheduled states
+        if flow_run.state.is_final() or flow_run.state.is_scheduled():
+            logger.debug(
+                f"Flow run {flow_run_id} is in final or scheduled state, skipping"
+            )
+            return
+
+        containers = event.get("detail", {}).get("containers", [])
+
+        containers_with_non_zero_exit_codes = [
+            container
+            for container in containers
+            if container.get("exitCode") is None or container.get("exitCode") != 0
+        ]
+
+        if any(containers_with_non_zero_exit_codes):
+            container_identifiers = [
+                c.get("name") or c.get("containerArn") for c in containers
+            ]
+            handler_logger.info(
+                "The following containers stopped with a non-zero exit code: %s. Marking flow run %s as crashed",
+                container_identifiers,
+                flow_run_id,
+            )
+            await propose_state(
+                client=orchestration_client,
+                state=Crashed(message="No active or succeeded pods found for any job"),
+                flow_run_id=uuid.UUID(flow_run_id),
+            )
+
+
+@ecs_observer.on_event(
+    "task",
+    tags={"prefect.io/degregister-task-definition": "true"},
+    statuses=["STOPPED"],
+)
+async def deregister_task_definition(event: dict[str, Any], tags: dict[str, str]):
+    handler_logger = logger.getChild("deregister_task_definition")
+
+    if not (task_definition_arn := event.get("detail", {}).get("taskDefinitionArn")):
+        handler_logger.debug("No task definition ARN in event. Skipping.")
+        return
+
+    async with aiobotocore.session.get_session().create_client("ecs") as ecs_client:
+        await ecs_client.deregister_task_definition(taskDefinition=task_definition_arn)
+        handler_logger.info(
+            "Task definition %s successfully deregistered", task_definition_arn
+        )
+
+
 _observer_task: asyncio.Task[None] | None = None
 
 
+def _observer_task_done(task: asyncio.Task[None]):
+    if task.cancelled():
+        logger.debug("ECS observer task cancelled")
+    elif task.exception():
+        logger.error("ECS observer task crashed", exc_info=task.exception())
+    else:
+        logger.debug("ECS observer task completed")
```

Comment:
Need to ensure the observer is restarted on exception. Otherwise messages backup in SQS.